shared/src/main/scala/io/kaitai/struct/languages/AwkwardCompiler.scala

```diff
@@ -723,7 +743,18 @@ class AwkwardCompiler(
           else
             outSrc.puts(s"auto& ${idToStr(id)}_builder = ${idToStr(id)}_listoffsetbuilder.content();")
           outSrc.puts(s"${idToStr(id)}_builder.append(${getRawIdExpr(id, rep)});")
-        case _: StrType | _: BytesType =>
+        case _: StrType =>
+          // Prints the C++ strings for appending the string data type to the Layoutbuilder.
+          var builderName = idToStr(id)
+          if (rep == NoRepeat)
+            outSrc.puts(s"auto& ${builderName}_stringbuilder = ${nameList.last}_builder.content<Field_${nameList.last}::${nameList.last + "A__Z" + idToStr(id)}>();")
+          else {
+            throw new NotImplementedError("StrType with repeat is not supported yet")
+            // builderName = "sub_" + builderName
```

Comment:
@ManasviGoyal I am not sure how to handle repetition, could you help me creating a version of `animal.ksy` with a repeated string, so I can use that for testing?

Reply:
I think we can leave this case as Not Implemented for now. As far as I know, it is not possible to have a repeat condition with a `str` type in a `ksy` file . It can be easily added later if we come across a case that requires it.